src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewCompiler.cs

```diff
@@ -100,17 +100,13 @@ public class RazorViewCompiler : IViewCompiler
             {
                 logger.ViewCompilerLocatedCompiledView(precompiledView.RelativePath);
 
-                if (_precompiledViews.TryGetValue(precompiledView.RelativePath, out var otherValue))
+                if (!_precompiledViews.ContainsKey(precompiledView.RelativePath))
                 {
-                    var message = string.Join(
-                        Environment.NewLine,
-                        Resources.RazorViewCompiler_ViewPathsDifferOnlyInCase,
-                        otherValue.RelativePath,
-                        precompiledView.RelativePath);
-                    throw new InvalidOperationException(message);
+                    // View ordering has precedence semantics, a view with a higher precedence was
+                    // already added to the list.
+                    _precompiledViews.Add(precompiledView.RelativePath, precompiledView);
```

Comment:
The RazorViewCompiler is responsible for precedence semantics. It uses the first view for a given path and ignores the rest.